samples/Common/cs/MediaPlayerFrameSource.cs

```diff
@@ -103,11 +121,21 @@ private void MediaPlayer_MediaOpened(MediaPlayer sender, object args)
         /// </summary>
         /// <param name="sender"></param>
         /// <param name="args"></param>
-        private void MediaPlayer_VideoFrameAvailable(MediaPlayer sender, object args)
+        private async void MediaPlayer_VideoFrameAvailable(MediaPlayer sender, object args)
         {
             m_mediaPlayer.CopyFrameToVideoSurface(m_videoFrame.Direct3DSurface);
-            m_videoFrame.SystemRelativeTime = m_mediaPlayer.PlaybackSession.Position;
-            FrameArrived?.Invoke(this, m_videoFrame);
+            
+            if(m_desiredImageDescriptor != null)
+            {
+                await m_videoFrame.CopyToAsync(m_stagingVideoFrame);
```

Comment:
Consider checking if source and desired format already match so we can avoid redundant copy if possible

Reply:
Not applicable if you just use m_videoFrame for all cases and remove m_stagingVideoFrame as suggested above